### ISSUE TITLE
Render process gone if a too big Json has been pasted into a new note

### ISSUE THREAD

**Issue by Wolle181** [laurent22/joplin#10564]
### Operating system

Windows

### Joplin version

2.14.22 (prod, win32)

### Desktop version info

Joplin **2.14.22** (prod, win32)

Client-ID: 10d96d0d427f45e3b9e098bd65721e5c
Sync-Version: 3
Profil-Version: 46
Unterstützter Schlüsselbund: Ja

Revision: e579eb9

Backup: 1.4.0

### Current behaviour

1. Open desktop app
2. Create a new note
3. In notepad++ or VSC, have a big json file in prety print format (in my case, 2 mio lines, ~60MB) and copy it into clipboard
4. In the new Joplin note, press Ctrl-V to paste it  

### Expected behaviour

Json will be shown or at least a message indicating "too big to show"
Instead: Shortly after pasting, a message box appears: "Render process gone" wiht optino to restart in safe mode.

Note: At my second attempt to reproduce this behaviour for this bug report, I could delete the too big note, but initially when this first happened, deletion was not possible even in safe mode. Deleting the .config dolder and re-sysnc solved the problem.

### Logs

[joplin_crash_dump_20240611T123833.json](https://github.com/user-attachments/files/15788099/joplin_crash_dump_20240611T123833.json)

The reproduction of the error occurred at 14:38:27 in the log ("textPaste").
[log.txt](https://github.com/user-attachments/files/15788114/log.txt)

**Comment by laurent22** [comment]
Does it work any better if you select the Beta editor from the options first?

**Comment by personalizedrefrigerator** [comment]
At least for me (beta editor), I can get Joplin to run very slowly after pasting Joplin's `yarn.lock` several times (24,803,840 characters), particularly when the note viewer is visible. I haven't tried with anything larger.

**Comment by Wolle181** [comment]
I retested this issue in version 3.0.13 with opted-in Beta Editor but it still did not come back really. Instead, a popup appeared which informed "Error occurred: Windows does not answer" (translated by me from German) and asked to either close, ignore or restart in safe mode. 

The good thing was that when restarting in safe mode, I was able to delete the new oversized entry which lead to the problem and after that could continue normally.

I think a solution could be restrict the size of the Json or switch off auto-formatting for too big content (as VSC does).

(The problem is not a major issue for me and I would prioritise it down. Anyone who has synchronised with WebDAV or made regular backups should be able to cope with it. )
